tests/unit/machine/IterativeMachine_unittest.cc

```diff
@@ -0,0 +1,82 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ */
+
+#include <gtest/gtest.h>
+
+#include <functional>
+#include <rxcpp/rx-lite.hpp>
+#include <shogun/lib/Signal.h>
+
+#include "environments/LinearTestEnvironment.h"
+#include "utils/SGObjectIterator.h"
+#include <shogun/base/some.h>
+#include <shogun/classifier/AveragedPerceptron.h>
+#include <shogun/classifier/Perceptron.h>
+#include <shogun/classifier/svm/NewtonSVM.h>
+#include <shogun/machine/IterativeMachine.h>
+#include <shogun/machine/LinearMachine.h>
+
+using namespace shogun;
+
+std::set<std::string> sg_linear_machines = {"Perceptron", "AveragedPerceptron",
+                                            "NewtonSVM"};
+extern LinearTestEnvironment* linear_test_env;
+
+TEST(IterativeMachine, continue_training_consistency)
+{
+	auto env = linear_test_env->getBinaryLabelData();
+	auto features = wrap(env->get_features_train());
+	auto labels = wrap(env->get_labels_train());
+	auto test_features = wrap(env->get_features_test());
+	auto test_labels = wrap(env->get_labels_test());
+
+	auto range = sg_object_iterator<untemplated_sgobject>(sg_linear_machines);
+	for (auto machine_obj : range)
+	{
+		const int32_t max_iters = 2;
+		auto machine = (CIterativeMachine<CLinearMachine>*)machine_obj;
+		auto machine_stop =
+		    (CIterativeMachine<CLinearMachine>*)machine->clone();
+		auto machine_iters =
+		    (CIterativeMachine<CLinearMachine>*)machine->clone();
+
+		// get result of fully trained machine
+		machine->set_labels(labels);
+		machine->train(features);
+		auto results = machine->apply(test_features);
+
+		index_t iter = 0;
+		auto callback = [&iter]() {
+			if (iter >= max_iters)
+			{
+				get_global_signal()->get_subscriber()->on_next(SG_BLOCK_COMP);
+				return true;
+			}
+			iter++;
+			return false;
+		};
+		machine_stop->set_callback(callback);
+		machine_stop->set_labels(labels);
+		machine_stop->train(features);
+
+		machine_iters->set_labels(labels);
+		machine_iters->put<int32_t>("max_iterations", max_iters);
```

Comment:
there should be yet a typed setter for this or?

Reply:
since this helps to make the tests for reliable (no runtime error when setting things as here), we can keep the setter (even though it will be hidden from users in the future)